runtime/lua/vim/uri.lua

```diff
@@ -110,10 +123,14 @@ function M.uri_to_fname(uri)
   end
   uri = M.uri_decode(uri)
   --TODO improve this.
-  if is_windows_file_uri(uri) then
+  if is_windows_volume_uri(uri) then
     uri = uri:gsub('^file:/+', ''):gsub('/', '\\')
   else
-    uri = uri:gsub('^file:/+', '/') ---@type string
+    if IS_WINDOWS and uri:match('^file://[a-zA-Z.-]') ~= nil then -- handle UNC file uri in windows
+      uri = uri:gsub('^file:/+', '//'):gsub('/', '\\')
```

Comment:
This is an example of logic that could be removed if we used `vim.normalize()` earlier in this function.

IOW: check for `file:` early in this function, and if true, use `vim.normalize()` on the rest of the path. What other logic would that allow us to remove, in uri.lua ?